code/obj/item/organ_holder.dm

```diff
@@ -713,6 +732,225 @@
 				return mytail
 
+	proc/drop_organ_augmentation(var/organ, var/location)
+		if (!src.donor || !organ)
+			return
+
+		if (!location)
+			location = src.donor.loc
+
+		if(!istext(organ) && istype(organ, /obj/item))
+			if(organ == head)
+				organ = "head"
+			else if(organ == brain)
+				organ = "brain"
+			else if(organ == left_eye)
+				organ = "left_eye"
+			else if(organ == right_eye)
+				organ = "right_eye"
+			else if(organ == heart)
+				organ = "heart"
+			else if(organ == left_lung)
+				organ = "left_lung"
+			else if(organ == right_lung)
+				organ = "right_lung"
+			else if(organ == left_kidney)
+				organ = "left_kidney"
+			else if(organ == right_kidney)
+				organ = "right_kidney"
+			else if(organ == liver)
+				organ = "liver"
+			else if(organ == stomach)
+				organ = "stomach"
+			else if(organ == intestines)
+				organ = "intestines"
+			else if(organ == spleen)
+				organ = "spleen"
+			else if(organ == pancreas)
+				organ = "pancreas"
+			else if(organ == appendix)
+				organ = "appendix"
+			else if(organ == tail)
+				organ = "tail"
+			else
+				return 0 // what the fuck are you trying to remove
+
+		switch(lowertext(organ))
+
+			if ("all")
+				if (islist(src.organ_list))
+					for (var/thing in src.organ_list)
+						if (!src.organ_list[thing])
+							continue
+						if (istype(thing, /obj/item/organ/chest)) //Chests don't and shouldn't support augmentations
+							continue
+						src.drop_organ_augmentation(thing, location)
+					return 1
+
+			if ("head") //returns nothing, use eyes/brain individually if you want returns
+				if (!src.head)
+					return 0
+				if (src.brain?.installed_aug)
+					src.brain.installed_aug = src.drop_organ_augmentation("brain")
+				if (src.right_eye?.installed_aug)
+					src.right_eye.installed_aug = src.drop_organ_augmentation("right_eye")
+				if (src.left_eye?.installed_aug)
+					src.left_eye.installed_aug = src.drop_organ_augmentation("left_eye")
+				src.donor.update_body()
+				src.donor.UpdateDamageIcon()
+				src.donor.update_clothing()
+				return 1
+
+			if ("brain")
+				if (!src.brain)
+					return 0
+				var/obj/item/augmentation/myAug = src.brain?.installed_aug
+				src.brain.installed_aug.set_loc(location)
+				src.brain.installed_aug.on_organ_removal()
+				src.brain.installed_aug = null
+				src.donor.update_body()
+				return myAug
+
+			if ("left_eye")
+				if (!src.left_eye)
+					return 0
+				var/obj/item/augmentation/myAug = src.left_eye?.installed_aug
+				src.left_eye.installed_aug.set_loc(location)
+				src.left_eye.installed_aug.on_organ_removal()
+				src.left_eye.installed_aug = null
+				src.donor.update_body()
+				return myAug
+
+			if ("right_eye")
+				if (!src.right_eye)
+					return 0
+				var/obj/item/augmentation/myAug = src.right_eye?.installed_aug
+				src.right_eye.installed_aug.set_loc(location)
+				src.right_eye.installed_aug.on_organ_removal()
+				src.right_eye.installed_aug = null
+				src.donor.update_body()
+				return myAug
```

Comment:
this is so much hell copypaste code, this doesn't need to be split up like this if the actual stuff being done is the same for everything here and below.